javalib/src/main/scala/java/math/BigDecimal.scala

```diff
@@ -672,13 +672,27 @@ class BigDecimal() extends Number with Comparable[BigDecimal] {
 
   def multiply(multiplicand: BigDecimal): BigDecimal = {
     val newScale = this._scale.toLong + multiplicand._scale
+
+    @inline
+    def bigMultiply: BigDecimal = {
+      val unscaled = this.getUnscaledValue.multiply(multiplicand.getUnscaledValue)
+      new BigDecimal(unscaled, safeLongToInt(newScale))
+    }
+
+    import java.lang.{Long => JLong}
     if (this.isZero || multiplicand.isZero) {
       zeroScaledBy(newScale)
-    } else if (this._bitLength + multiplicand._bitLength < 64) {
-      valueOf(this._smallValue * multiplicand._smallValue, safeLongToInt(newScale))
+    } else if (this._bitLength + multiplicand._bitLength >= 64) {
+      bigMultiply
     } else {
-      val unscaled = this.getUnscaledValue.multiply(multiplicand.getUnscaledValue)
-      new BigDecimal(unscaled, safeLongToInt(newScale))
+      val x = this._smallValue
+      val y = multiplicand._smallValue
+      val z = x * y
+      if (JLong.signum(z) == JLong.signum(x) * JLong.signum(y)) {
```

Comment:
Is there any other case than `MinValue * MinValue` here?

If not, then explicitly testing `x != MinValue || y != MinValue` would make that clearer, and be faster (long multiplication is mildly expensive in Scala.js)

If yes, could you give another example that needs this test like this? Why does this test always work for other problematic situations? (unlike for addition, in 2's complement multiplication checking the sign of the result is not enough to detect overflow in general).